pom.xml

```diff
@@ -2933,9 +2933,9 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
-              <meminitial>1024m</meminitial>
-              <maxmem>8192m</maxmem>
+              <failOnError>true</failOnError>
```

Comment:
How will the maven execution fail without this?

Reply:
If I understood it right, the final commit must have the changes of groovy including `failOnError` be true for all checkers.

Reply:
> If I understood it right, the final commit must have the changes of groovy including `failOnError` be true for all checkers.

Yes, that’s right. Right now, we only have one updated, and it is in the “hack” to cause OOM commit.

Reply:
Thank you for the review. I have made the changes. Please let me know if they are right.

Reply:
Ok, let's drop other commits and prepare for merge.

Reply:
> Ok, let's drop other commits and prepare for merge.

Done.

Reply:
@nrmancuso, I couldn't find anything substantial in the CI failures. Can you help me point what is leading to the failure?

Reply:
@relentless-pursuit codenarc is failing in semaphore:
`Violation: Rule=ThrowRuntimeException P=2 Line=132 Msg=[The type RuntimeException should not be thrown] Src=[throw new RuntimeException("Maven process exited with error code: " + exitCode)]`

Reply:
> `System.exit(1)`

After thinking on this, this is a hack that we should probably avoid. I also have doubts about whether we would execute the `finally` block below this to close the reader. @relentless-pursuit let's keep it the way you had it and find a more specific exception class as recommended by codenarc: https://codenarc.org/codenarc-rules-exceptions.html#:~:text=Checks%20for%20throwing%20an%20instance%20of%20java.lang.RuntimeException.%20Throw%20an%20instance%20of%20a%20more%20specific%20exception%20subclass%20instead.

Reply:
> https://codenarc.org/codenarc-rules-exceptions.html#:~:text=Checks%20for%20throwing%20an%20instance%20of%20java.lang.RuntimeException.%20Throw%20an%20instance%20of%20a%20more%20specific%20exception%20subclass%20instead

Sure, I was cross-checking and it didn't give me OOM error.

Reply:
Let's do `IllegalStateException`

Reply:
> Let's do `IllegalStateException`

Done. I also checked with the hack, and it seems to be working fine. https://github.com/checkstyle/checkstyle/actions/runs/7974597699/job/21770876534?pr=14409